spec/classes/foreman_proxy__plugin__openscap_spec.rb

```diff
@@ -0,0 +1,41 @@
+require 'spec_helper'
+
+describe 'foreman_proxy::plugin::openscap' do
+  on_supported_os.each do |os, facts|
+    context "on #{os}" do
+      context 'openscap plugin is enabled' do
+        let :params do
+          {
+            :enabled => true
+          }
+        end
+
+        it 'should call the plugin' do
+          should contain_foreman_proxy__plugin('openscap')
+        end
+
+        it 'should install configuration file' do
+          should contain_foreman_proxy__settings_file('openscap')
+          should contain_file('/etc/foreman-proxy/settings.d/openscap.yml').with({:content => /:enabled: https/})
+        end
+      end
+
+      context 'openscap plugin is disabled' do
+        let :params do
+          {
+            :enabled => false
+          }
+        end
+
+        it 'should call the plugin' do
+          should contain_foreman_proxy__plugin('openscap')
+        end
+
+        it 'should install configuration file' do
+          should contain_foreman_proxy__settings_file('openscap')
+          should contain_file('/etc/foreman-proxy/settings.d/openscap.yml').with({:content => /:enabled: false/})
```

Comment:
I'd also prefer to test the full content here. https://github.com/theforeman/puppet-foreman_proxy/blob/49435dc0d8a63c779f666a193e3205423644ad46/spec/classes/foreman_proxy__config__spec.rb#L63-L80 is such an example.